community/detectors/bitbucket_cve_2022_36804/src/main/java/com/google/tsunami/plugins/detectors/bitbucket/Cve202236804VulnDetector.java

```diff
@@ -0,0 +1,212 @@
+        bootstrapModule = Cve202236804DetectorBootstrapModule.class)
+public class Cve202236804VulnDetector implements VulnDetector {
+
+    private static final GoogleLogger logger = GoogleLogger.forEnclosingClass();
+    private static final String GET_ALL_PUB_PATH =
+            "repos?visibility=public";
+    private static final String STRING_PUB_REP = "Public Repositories";
+    private final HttpClient httpClient;
+    private final Clock utcClock;
+    private final PayloadGenerator payloadGenerator;
+
+    @Inject
+    Cve202236804VulnDetector(@UtcClock Clock utcClock, HttpClient httpClient, PayloadGenerator payloadGenerator) {
+        this.httpClient = checkNotNull(httpClient).modify().setFollowRedirects(false).build();
+        this.utcClock = checkNotNull(utcClock);
+        this.payloadGenerator = checkNotNull(payloadGenerator);
+    }
+
+    private static boolean isWebServiceOrUnknownService(NetworkService networkService) {
+        return networkService.getServiceName().isEmpty()
+                || NetworkServiceUtils.isWebService(networkService);
+    }
+
+    private static String buildTargetUrl(NetworkService networkService, String url) {
+        StringBuilder targetUrlBuilder = new StringBuilder();
+        if (NetworkServiceUtils.isWebService(networkService)) {
+            targetUrlBuilder.append(NetworkServiceUtils.buildWebApplicationRootUrl(networkService));
+        } else {
+            // Assume the service uses HTTP protocol when the scanner cannot identify the actual service.
+            targetUrlBuilder
+                    .append("http://")
+                    .append(toUriAuthority(networkService.getNetworkEndpoint()))
+                    .append("/");
+        }
+        targetUrlBuilder.append(url);
+        return targetUrlBuilder.toString();
+    }
+
+    private boolean isServiceVulnerable(NetworkService networkService) {
+
+        PayloadGeneratorConfig config =
+                PayloadGeneratorConfig.newBuilder()
+                        .setVulnerabilityType(PayloadGeneratorConfig.VulnerabilityType.BLIND_RCE)
+                        .setInterpretationEnvironment(
+                                PayloadGeneratorConfig.InterpretationEnvironment.LINUX_SHELL)
+                        .setExecutionEnvironment(
+                                PayloadGeneratorConfig.ExecutionEnvironment.EXEC_INTERPRETATION_ENVIRONMENT)
+                        .build();
+
+        if (!payloadGenerator.isCallbackServerEnabled()) {
+            return false;
+        }
+        Payload payload = this.payloadGenerator.generate(config);
+
+        String commandToInject = String.format("sh -c \"%s\"", payload.getPayload());
+
+        String PubRepUrl = buildTargetUrl(networkService, GET_ALL_PUB_PATH);
+
+        try {
+            HttpResponse httpResponse =
+                    httpClient.send(get(PubRepUrl).withEmptyHeaders().build(), networkService);
+            if (httpResponse.status().code() == 200
```

Comment:
Please prefer an early return approach to make the code more readable. In other words:

if (code != 200) || !contains(STRING_PUB_REP) {
  return false;
}

// continue detection